### DISCUSSION TITLE
How to speed up plotting massive data and interaction with the plot?

### DISCUSSION THREAD

**Discussion by wanggy-1** [pyvista/pyvista#1697] [Q&A] [answered]
Hello,
I'm trying to plot a 500x500x500 (xyz) geological model with pyvista.StructureGrid(), but it's really slow when visualizing this model with add_mesh(), and it responded really slow when I try to zoom or rotate the model.
Any method to speed up plotting and interaction?
Thanks.

**Comment by banesullivan** [comment, marked as answer]
Can you share the data? Can you share a screenshot at least?

Does the model have a uniform structure? My first suspicion is that `StructuredGrid` might not be the right data type and a `UniformGrid` would work. `UniformGrid`'s are much more performant. 

However, if a `StructuredGrid` is indeed the right choice because there is some sort of curvilinear structure, then this is more likely a hardware issue than a software issue

Here is an example with a grid of that size. Creating it takes a moment, but plotting is quite smooth on my machine which is pretty spec'ed out:

```py
import numpy as np
import pyvista as pv

x = y = z = np.arange(500)
xx, yy, zz = np.meshgrid(x, y, z)

grid = pv.StructuredGrid(xx, yy, zz)
print(grid.dimensions)

p = pv.Plotter(notebook=False)
p.add_mesh(grid)
p.show()
```

![2021-11-19 11 09 03](https://user-images.githubusercontent.com/22067021/142670862-91017d6c-7cdf-4207-8221-b3151e27b849.gif)

**Reply by banesullivan** [reply]
Having a dedicated GPU is pretty much necessary

**Reply by wanggy-1** [reply]
Yes, you're right. I was using the CPU as renderer before. Now I use the GPU as renderer and the plotting is smooth just like your demonstration. Thanks.
